tools/model-apt/src/main/java/io/smallrye/openapi/model/apt/SmallRyeModelProcessor.java

```diff
@@ -0,0 +1,743 @@
+                    return buffer.toString();
+                });
+
+        if (property.singularName != null) {
+            buffer.setLength(0);
+            buffer.append(property.singularName);
+            buffer.setCharAt(0, Character.toUpperCase(property.singularName.charAt(0)));
+            property.singularMethodName = buffer.toString();
+        }
+
+        return property;
+    }
+
+    private String getSingularName(String propertyName, Map<String, Object> values) {
+        String singularName = get(values, "singularName", String.class);
+
+        if (singularName.isBlank()) {
+            if (propertyName.endsWith("s")) {
+                singularName = propertyName.substring(0, propertyName.length() - 1);
+            } else {
+                throw new IllegalArgumentException(
+                        "singularName is required for map or list type when name does not end with 's': " + propertyName);
+            }
+        }
+
+        return singularName;
+    }
+
+    private void writeProperty(Writer writer, String className, PropertyInfo property)
+            throws IOException {
+
+        if (isCentralizedProperty(property)) {
+            return;
+        }
+
+        writeGet(writer, property);
+        writeSet(writer, property);
+
+        if (Map.class.getName().equals(property.rawType)) {
+            writeMapAdd(writer, className, property);
+            writeMapRemove(writer, property);
+        } else if (List.class.getName().equals(property.rawType)) {
+            writeListAdd(writer, className, property);
+            writeListRemove(writer, property);
+        }
+    }
+
+    /**
+     *
+     * @param writer
+     * @param property
+     * @throws IOException
+     */
+    private void writeGet(Writer writer, PropertyInfo property) throws IOException {
+        writeJavaDoc(writer);
+        writeOverride(writer);
+
+        writeCodeLn(writer, 1, PUBLIC, property.type, " get", property.methodName, "() {");
+
+        if (property.unwrapped) {
+            String valueTypeParam;
+
+            if (property.valueType.indexOf('<') > -1) {
+                /*
+                 * We only care about the raw type, but the compiler needs to be fooled to obtain
+                 * the type with generic arguments to satisfy the call to getProperties, and the
+                 * return type.
+                 */
+                String rawType = property.valueType.substring(0, property.valueType.indexOf('<'));
+                writeCodeLn(writer, 2, "@SuppressWarnings(\"unchecked\")");
+                writeCodeLn(writer, 2, "Class<", property.valueType, "> TYPE = (Class<", property.valueType, ">) (Object) ",
+                        rawType,
+                        CLASS, ";");
+                valueTypeParam = "TYPE";
+            } else {
+                valueTypeParam = property.valueType + CLASS;
+            }
+
+            writeCodeLn(writer, 2, "return getProperties(", valueTypeParam, ");");
```

Comment:
This change means that some methods which could previously return `null` will instead return an empty map.

- `APIResponses.getAPIResponses()`
- `Content.getMediaTypes()`
- `Callback.getPathItems()`
- `Paths.getPathItems()`
- `SecurityRequirement.getSchemes()`

I'm unsure if this is worth worrying about, but it is probably worth noting as a change in behaviour since 3.x.

Reply:
Right, these objects are quasi-maps themselves so non-null seems to make sense there if the map returned is a map form of the object itself.